apps/prairielearn/src/lib/course-request.sql

```diff
@@ -69,43 +69,13 @@ WITH
       INNER JOIN users AS u ON u.user_id = r.user_id
       LEFT JOIN users AS ua on ua.user_id = r.approved_by
       LEFT JOIN select_course_request_jobs AS j ON j.id = r.id
-  ),
-  select_institutions_with_authn_providers AS (
-    SELECT
-      i.*,
-      coalesce(
-        jsonb_agg(
-          ap.name
-          ORDER BY
-            ap.name
-        ),
-        '[]'::jsonb
-      ) AS authn_providers
-    FROM
-      institutions AS i
-      LEFT JOIN institution_authn_providers AS iap ON (iap.institution_id = i.id)
-      LEFT JOIN authn_providers AS ap ON (ap.id = iap.authn_provider_id)
-    GROUP BY
-      i.id
-  ),
-  select_institutions AS (
-    SELECT
-      coalesce(
-        jsonb_agg(
-          i
-          ORDER BY
-            i.short_name
-        ),
-        '[]'::jsonb
-      ) AS institutions
-    FROM
-      select_institutions_with_authn_providers AS i
+    WHERE
+      $show_all = 'true'
+      OR r.approved_status NOT IN ('approved', 'denied')
   )
 SELECT
```

Comment:
Final comment: AFAICT, you don't need to separate the selection into a CTE and this `SELECT ...`. That is, you can replace the `select_course_requests AS ...` CTE with just the contents of the CTE instead (the `SELECT coalesce(...` bit inside the `select_course_requests` CTE). Let me know if that makes sense! You'd then have to update `selectCourseRequests` to use `queryRows` with `CourseRequestRowSchema` instead of `z.array(CourseRequestRowSchema)`.

Once you've done that, there's also no more reason to pre-aggregate the course requests into a JSON array; you can just return the columns directly.

To spell this all out a bit, here's how this query could look:

```sql
WITH
  select_course_request_jobs AS (
    ...
  )
SELECT
  r.id,
  r.short_name,
  ...
FROM
  course_requests AS r ...
```

Let me know if this makes sense or if you'd like additional clarification.

For context: the only reason this query was originally written differently was because we wanted to return lists of both institutions and course requests from the same query. Now that we're returning just a homogenous list of course requests, we can allow each course request to be a separate row with standard columns.

Reply:
That all makes sense! I have reduced this query down in [fc2c86b](https://github.com/PrairieLearn/PrairieLearn/pull/9238/commits/fc2c86b156680165f9b4bf1034bdadc35f08a01b).